### ISSUE TITLE
Incompletely iterated scanners going out of scope are not cleaned

### ISSUE THREAD

**Issue by wbolster** [wbolster/happybase#8]
If the iterator returned by Table.scan() is not completely exhausted (i.e. due to a `break` in a `for` loop), HappyBase does not close the server-side scanner, even if the Table.scan() result (currently a generator function) has gone out of scope. The (leaked) associated server-side resources are freed only after the scanner times out.

A possible solution would be to make the return type from Table.scan() a real class with `__iter__()` and `next()` functions and a `__del__()` to make sure resources are freed as soon as possible. Additionally, this class can be made a context manager, so that something like this can be used for scanners that may not be fully iterated over:

``` python
with table.scan() as scan:
    for row_key, row_data in scan:
        pass  # do something, possibly breaking out of the loop
```
